### PR TITLE
Update volume scheduling design with event handling details.

### PR DESCRIPTION
**Pull request by cofyc**
Fixes https://github.com/kubernetes/kubernetes/issues/60002.

This PR addresses volume topology scheduling related predicates only.

Related issues and prs:

- https://github.com/kubernetes/kubernetes/pull/55039
- https://github.com/kubernetes/kubernetes/issues/58797
- https://github.com/kubernetes/kubernetes/pull/59335
- https://github.com/kubernetes/kubernetes/pull/66106
- https://github.com/kubernetes/kubernetes/pull/65616

### CODE REVIEW

File: contributors/design-proposals/storage/volume-topology-scheduling.md [hunk 1]
```diff
@@ -862,6 +862,13 @@ processing later in the priority and bind functions.
    temporarily cache this decision in the PVC per Node.
 8. Otherwise return false.
 
+Update NoVolumeZoneConflict predicate function to skip unbound PVC if StorageClass binding
```
Comment (msau42) [inline contributors/design-proposals/storage/volume-topology-scheduling.md]: This may fit better under the "Other predicates" section

Comment (cofyc) [reply contributors/design-proposals/storage/volume-topology-scheduling.md]: Updated.

**Comment by msau42** [comment]
/lgtm

Thanks for updating this!

**Comment by cofyc** [comment]
cc @saad-ali for approval

**Pull request merged**